### DISCUSSION TITLE
Is getServerSideProps the *only* way to have dynamic meta tags?

### DISCUSSION THREAD

**Discussion by Kae7in** [vercel/next.js#22789] [Help]
We need dynamic meta tags (different image, title, description) for each detail page and each user profile. Is rendering via getServerSideProps the ONLY way to make this happen? — because it is considerably slower than client-side rendering.

The main reason I don't think we can use getStaticProps is because our hosting service (Vercel) has a 16,000 static build files output limit. That's not enough, plus I assume builds would take considerably longer.

Any tips?

**Comment by leerob** [comment]
Hey! No, it's not the only way. You can also switch out meta tags on the client-side.

Re: Vercel. Rather than building 16,000 files at build-time, have you considered using `fallback` with `getStaticProps`? This will keep your build times low, as you can specify specifically how many pages you want to statically generate at build time - and then the request can be generated on the fly with `fallback`. 

https://nextjs.org/docs/basic-features/data-fetching#fallback-true
